apps/api/src/app/routes/__chainId/simulation/simulateBundle.ts

```diff
@@ -130,23 +130,32 @@ const root: FastifyPluginAsync = async (fastify): Promise<void> => {
       },
     },
     async function (request, reply) {
-      const { chainId } = request.params;
+      try {
+        const { chainId } = request.params;
 
-      const simulationResult =
-        await tenderlyService.postTenderlyBundleSimulation(
-          chainId,
-          request.body
+        fastify.log.info(
+          `Starting simulation of ${request.body.length} transactions on chain ${chainId}`
         );
 
-      if (simulationResult === null) {
-        reply.code(400).send({ message: 'Build simulation error' });
-        return;
-      }
-      fastify.log.info(
-        `Post Tenderly bundle of ${request.body.length} simulation on chain ${chainId}`
-      );
+        const simulationResult =
+          await tenderlyService.postTenderlyBundleSimulation(
+            chainId,
+            request.body
+          );
+
+        if (simulationResult === null) {
+          reply.code(400).send({ message: 'Build simulation error' });
```

Comment:
This was here before, but what does `null` mean, and why is a `400` is an issue from the user?

Reply:
It means that the Tenderly couldn't simulate the transaction. This happens often on the "build your own hook" where the user manually encodes the tx data.

We used a general 400 error but maybe we could change it to 422.